mirrord/layer/src/debugger_ports.rs

```diff
@@ -33,34 +57,47 @@ impl FromStr for DebuggerType {
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match s {
```

Comment:
looking at this PR, I think we shouldn't have a variant for each `DebuggerType` since the detection and port extracting logic is quite similar and repeated..
A good question would be why not have the enum's impl as a standalone function that just returns port to ignore? holding the variant doens't help us anyway?

Reply:
to be more concrete - I am afraid that the detection here is too "lite" while just trying to extract the port is way more full

Reply:
> to be more concrete - I am afraid that the detection here is too "lite" while just trying to extract the port is way more full

I'm not sure if I follow. When we use this feature we know exactly which debugger is used.

Reply:
I am saying we shouldn't split the logic into "detection" and "extraction" - just try to extract and if it fails fallback to env port and if that is missing/fails don't ignore anything.

Reply:
instead of 
```rs
let debugger_type = "pydevd".parse()
let ignore_port = debugger_type.get_port()
```
do
```rs
let ignore_port = get_debugger_ignore_port()

fn get_debugge_ignore_port() -> Option<u16> {    
  if python {
        if pydevd {}
        else if otherthingy {}
        else { envport}
    } else { envport} 
}

Reply:
Hmm, this detection thing was added having a specific case in mind - when the debuggers are invoked by the IDEs and the users have no way to select the port themselves. If the user has control over the port selection, I'd rather leave it up to them to explicitly ignore it in the config (or not, if that is their use case). But if you think this would improve UX, I can change it

Reply:
Resolving after a call :heavy_check_mark: